lib/features/sks-menu/presentation/widgets/sks_menu_view_loading.dart

```diff
@@ -0,0 +1,53 @@
+import "package:flutter/material.dart";
+
+import "../../../../config/ui_config.dart";
+import "../../../../theme/app_theme.dart";
+import "../../../../widgets/loading_widgets/shimmer_loading.dart";
+import "sks_menu_header.dart";
+import "sks_menu_tiles.dart";
+
+class SksMenuViewLoading extends StatelessWidget {
+  const SksMenuViewLoading({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: context.colorTheme.whiteSoap,
+      body: Shimmer(
+        linearGradient: shimmerGradient,
+        child: ListView(
+          physics: const NeverScrollableScrollPhysics(),
+          children: [
+            const ShimmerLoadingItem(
+              child: SksMenuHeaderLoading(),
+            ),
+            sksMenuTitleLoadingPaddings(),
+            sksMenuTitleLoadingPaddings(),
+            sksMenuTitleLoadingPaddings(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Padding sksMenuTitleLoadingPaddings() {
```

Comment:
so this is called "builder methods" and its a flutter anti-pattern (bad thing to do).
Instead we should always opt for creating a separate widget class. Article on a topic: https://medium.com/flutter-community/splitting-widgets-to-methods-is-a-performance-antipattern-16aa3fb4026c

Reply:
<img width="504" alt="Zrzut ekranu 2024-11-24 o 16 00 54" src="https://github.com/user-attachments/assets/bbb2b032-9ec4-434f-9ca4-e3abcc782196">